src/adr/0011-remove-elasticsearch-document-source.md

```diff
@@ -0,0 +1,33 @@
+# 11. Disable ElasticSearch source
+
+Date: 2019-10-17
+
+## Status
+
+Rejected
+
+The benefits do not outweigh the costs.
+
+## Context
+
+Though very handy to have around, the source field does incur storage overhead within the index. 
+
+## Decision
+
+Disable \_source for ELasticSearch indexed documents.
+
+## Consequences
+
+Given a dataset composed of small text/plain messages, we notice a 20% space reduction of data stored on ElasticSearch.
+
+However, patch updates can no longer be performed upon flags updates. Upon flag update we need to fully read the mail 
```

Comment:
Unless you choose another indexing structure

What about indexing flags separately from messages?
I could also prevent the multiple indexation of a given message when it is present in several mailboxes, don't you think?
It could also be a good idea to store differently frequently changing data (flags, mailbox) from almost constant data (the mail content).

Reply:
Parent-child relationships comes with it's own sets of problems. Parent on child needs to live on the same index, thus needs to have similar structure.

Also we need evaluation of the "JOIN" cost upon flags searches/filtering.

Such a proposal impacts _source decision, but is beneficial even outside this context. We can rechallenge the _source debate once your proposal is implemented.

Note please that _source allows admin flexibility like **performing the following operations without downtime**:
 - Increase shards
 - Modifying replication factor
 - Changing analysers (IE allows an admin to configure FR analyser instead of EN analyser)

Amongst others.

If we *ever* consider removing the **_source** field, we significantly harm possible admin maintainance over our work. I find it hard to push it for just a 20% size enhancement. I would really prefer exposing an option in the James configuration to state if we want to use **_source** or not.